### ISSUE TITLE
Using relative path for compiler in compiler_commands.json breaks intellisense

### ISSUE THREAD

**Issue by allight** [microsoft/vscode-cpptools#3112]
Issue Type: <b>Bug</b>

Using a compile_commands.json which has the compiler as a path relative to the 'directory' entry causes completion and other intellisense features to fail.

Repro:

1) Download this example git project: https://github.com/allight/compile-commands-example (`git clone https://github.com/allight/compile-commands-example.git`)

2) run `./gen_compile_commands.json`

3) Open vscode

4) try to use hover on (for example) line 23.

You can generate a compile_commands.json which will work correctly by passing --working to gen_compile_commands.py. That changes the compiler path to an absolute path.

Expected:

A pop-up giving the name and comment for the `theAnswer` function appears.

Result:

Box forever shows `loading...`

You can also see this if you try to use vscode on AOSP, you will need to revert this commit https://android.googlesource.com/platform/build/soong/+/e81281865672b80c21dfbdfbc07246335caa0186 (`cd build/soong; git revert e81281865672b80c21dfbdfbc07246335caa0186`). Creating the compile_commands.json file can be done by running `. build/envsetup.sh; m SOONG_GEN_COMPDB=1 SOONG_LINK_COMPDB_TO=$ANDROID_BUILD_TOP nothing`

Output from 'Debug' log level with failing `compile_commands.json`:

```
initialized
workspace/didChangeConfiguration
IntelliSense Engine = Default.
The extension will use the Tag Parser for IntelliSense when #includes don't resolve.
Autocomplete is enabled.
Error squiggles are enabled.
File exclude: **/.git
File exclude: **/.svn
File exclude: **/.hg
File exclude: **/CVS
File exclude: **/.DS_Store
File exclude: **/.repo
File exclude: **/.vscode
Search exclude: **/node_modules
Search exclude: **/bower_components
Search exclude: **/.vscode
textDocument/didOpen
cpptools/queryCompilerDefaults
Attempting to get defaults from compiler found on the machine: '/usr/bin/clang'
terminating child process: 41091
terminating child process: 41102
Attempting to get defaults from compiler found on the machine: '/usr/bin/clang'
terminating child process: 41119
cpptools/activeDocumentChange
cpptools/textEditorSelectionChange
textDocument/documentSymbol
cpptools/didChangeFolderSettings
Code browsing service initialized
  Folder: /usr/lib/llvm-4.0/lib/clang/4.0.1/include/ will be indexed
  Folder: /usr/local/include/ will be indexed
  Folder: /usr/include/ will be indexed
  Folder: /home/allight/misc/compile-commands-example/ will be indexed
textDocument/didOpen
Populate include completion cache.
Attempting to get defaults from compiler in "compilerPath" property: 'prebuilts/my-clang'
```

Expected output (using compile_commands.json from `./gen_compile_commands.py --working`):

```
initialized
workspace/didChangeConfiguration
IntelliSense Engine = Default.
The extension will use the Tag Parser for IntelliSense when #includes don't resolve.
Autocomplete is enabled.
Error squiggles are enabled.
File exclude: **/.git
File exclude: **/.svn
File exclude: **/.hg
File exclude: **/CVS
File exclude: **/.DS_Store
File exclude: **/.repo
File exclude: **/.vscode
Search exclude: **/node_modules
Search exclude: **/bower_components
Search exclude: **/.vscode
cpptools/queryCompilerDefaults
Attempting to get defaults from compiler found on the machine: '/usr/bin/clang'
terminating child process: 43247
terminating child process: 43260
Attempting to get defaults from compiler found on the machine: '/usr/bin/clang'
terminating child process: 43262
cpptools/activeDocumentChange
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
textDocument/didOpen
textDocument/documentSymbol
textDocument/codeAction
cpptools/didChangeFolderSettings
Code browsing service initialized
  Folder: /usr/lib/llvm-4.0/lib/clang/4.0.1/include/ will be indexed
  Folder: /usr/local/include/ will be indexed
  Folder: /usr/include/ will be indexed
  Folder: /home/allight/misc/compile-commands-example/ will be indexed
textDocument/didOpen
Populate include completion cache.
Attempting to get defaults from compiler in "compilerPath" property: '/home/allight/misc/compile-commands-example/prebuilts/my-clang'
terminating child process: 43264
terminating child process: 43266
Discovering files...
textDocument/documentSymbol
sending compilation args for /home/allight/misc/compile-commands-example/test.cc
  include: /usr/include/c++
  include: /usr/lib/llvm-4.0/lib/clang/4.0.1/include
  include: /usr/local/include
  include: /usr/include/x86_64-linux-gnu
  include: /usr/include
  define: __llvm__=1
  define: __clang__=1
  define: __clang_major__=4
  define: __clang_minor__=0
  define: __clang_patchlevel__=1
  define: __clang_version__="4.0.1 (tags/RELEASE_401/final)"
  define: __GNUC_MINOR__=2
  define: __GNUC_PATCHLEVEL__=1
  define: __GNUC__=4
  define: __GXX_ABI_VERSION=1002
  define: __ATOMIC_RELAXED=0
  define: __ATOMIC_CONSUME=1
  define: __ATOMIC_ACQUIRE=2
  define: __ATOMIC_RELEASE=3
  define: __ATOMIC_ACQ_REL=4
  define: __ATOMIC_SEQ_CST=5
  define: __PRAGMA_REDEFINE_EXTNAME=1
  define: __VERSION__="4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)"
  define: __STRICT_ANSI__=1
  define: __GXX_EXPERIMENTAL_CXX0X__=1
  define: __cpp_rtti=199711
  define: __cpp_exceptions=199711
  define: __cpp_unicode_characters=200704
  define: __cpp_raw_strings=200710
  define: __cpp_unicode_literals=200710
  define: __cpp_user_defined_literals=200809
  define: __cpp_lambdas=200907
  define: __cpp_constexpr=201304
  define: __cpp_range_based_for=201603
  define: __cpp_static_assert=201411
  define: __cpp_decltype=200707
  define: __cpp_attributes=200809
  define: __cpp_rvalue_references=200610
  define: __cpp_variadic_templates=200704
  define: __cpp_initializer_lists=200806
  define: __cpp_delegating_constructors=200604
  define: __cpp_nsdmi=200809
  define: __cpp_inheriting_constructors=201511
  define: __cpp_ref_qualifiers=200710
  define: __cpp_alias_templates=200704
  define: __cpp_binary_literals=201304
  define: __cpp_digit_separators=201309
  define: __cpp_init_captures=201304
  define: __cpp_generic_lambdas=201304
  define: __cpp_decltype_auto=201304
  define: __cpp_return_type_deduction=201304
  define: __cpp_aggregate_nsdmi=201304
  define: __cpp_variable_templates=201304
  define: __cpp_hex_float=201603
  define: __cpp_inline_variables=201606
  define: __cpp_noexcept_function_type=201510
  define: __cpp_capture_star_this=201603
  define: __cpp_if_constexpr=201606
  define: __cpp_template_auto=201606
  define: __cpp_namespace_attributes=201411
  define: __cpp_enumerator_attributes=201411
  define: __cpp_nested_namespace_definitions=201411
  define: __cpp_variadic_using=201611
  define: __cpp_aggregate_bases=201603
  define: __cpp_structured_bindings=201606
  define: __cpp_nontype_template_args=201411
  define: __cpp_fold_expressions=201603
  define: __cpp_aligned_new=201606
  define: __CONSTANT_CFSTRINGS__=1
  define: __EXCEPTIONS=1
  define: __GXX_RTTI=1
  define: __DEPRECATED=1
  define: __GNUG__=4
  define: __GXX_WEAK__=1
  define: __private_extern__=extern
  define: __ORDER_LITTLE_ENDIAN__=1234
  define: __ORDER_BIG_ENDIAN__=4321
  define: __ORDER_PDP_ENDIAN__=3412
  define: __BYTE_ORDER__=__ORDER_LITTLE_ENDIAN__
  define: __LITTLE_ENDIAN__=1
  define: _LP64=1
  define: __LP64__=1
  define: __CHAR_BIT__=8
  define: __SCHAR_MAX__=127
  define: __SHRT_MAX__=32767
  define: __INT_MAX__=2147483647
  define: __LONG_MAX__=9223372036854775807L
  define: __LONG_LONG_MAX__=9223372036854775807LL
  define: __WCHAR_MAX__=2147483647
  define: __INTMAX_MAX__=9223372036854775807L
  define: __SIZE_MAX__=18446744073709551615UL
  define: __UINTMAX_MAX__=18446744073709551615UL
  define: __PTRDIFF_MAX__=9223372036854775807L
  define: __INTPTR_MAX__=9223372036854775807L
  define: __UINTPTR_MAX__=18446744073709551615UL
  define: __SIZEOF_DOUBLE__=8
  define: __SIZEOF_FLOAT__=4
  define: __SIZEOF_INT__=4
  define: __SIZEOF_LONG__=8
  define: __SIZEOF_LONG_DOUBLE__=16
  define: __SIZEOF_LONG_LONG__=8
  define: __SIZEOF_POINTER__=8
  define: __SIZEOF_SHORT__=2
  define: __SIZEOF_PTRDIFF_T__=8
  define: __SIZEOF_SIZE_T__=8
  define: __SIZEOF_WCHAR_T__=4
  define: __SIZEOF_WINT_T__=4
  define: __SIZEOF_INT128__=16
  define: __INTMAX_TYPE__=long int
  define: __INTMAX_FMTd__="ld"
  define: __INTMAX_FMTi__="li"
  define: __INTMAX_C_SUFFIX__=L
  define: __UINTMAX_TYPE__=long unsigned int
  define: __UINTMAX_FMTo__="lo"
  define: __UINTMAX_FMTu__="lu"
  define: __UINTMAX_FMTx__="lx"
  define: __UINTMAX_FMTX__="lX"
  define: __UINTMAX_C_SUFFIX__=UL
  define: __INTMAX_WIDTH__=64
  define: __PTRDIFF_TYPE__=long int
  define: __PTRDIFF_FMTd__="ld"
  define: __PTRDIFF_FMTi__="li"
  define: __PTRDIFF_WIDTH__=64
  define: __INTPTR_TYPE__=long int
  define: __INTPTR_FMTd__="ld"
  define: __INTPTR_FMTi__="li"
  define: __INTPTR_WIDTH__=64
  define: __SIZE_TYPE__=long unsigned int
  define: __SIZE_FMTo__="lo"
  define: __SIZE_FMTu__="lu"
  define: __SIZE_FMTx__="lx"
  define: __SIZE_FMTX__="lX"
  define: __SIZE_WIDTH__=64
  define: __WCHAR_TYPE__=int
  define: __WCHAR_WIDTH__=32
  define: __WINT_TYPE__=unsigned int
  define: __WINT_WIDTH__=32
  define: __SIG_ATOMIC_WIDTH__=32
  define: __SIG_ATOMIC_MAX__=2147483647
  define: __CHAR16_TYPE__=unsigned short
  define: __CHAR32_TYPE__=unsigned int
  define: __UINTMAX_WIDTH__=64
  define: __UINTPTR_TYPE__=long unsigned int
  define: __UINTPTR_FMTo__="lo"
  define: __UINTPTR_FMTu__="lu"
  define: __UINTPTR_FMTx__="lx"
  define: __UINTPTR_FMTX__="lX"
  define: __UINTPTR_WIDTH__=64
  define: __FLT_DENORM_MIN__=1.40129846e-45F
  define: __FLT_HAS_DENORM__=1
  define: __FLT_DIG__=6
  define: __FLT_DECIMAL_DIG__=9
  define: __FLT_EPSILON__=1.19209290e-7F
  define: __FLT_HAS_INFINITY__=1
  define: __FLT_HAS_QUIET_NAN__=1
  define: __FLT_MANT_DIG__=24
  define: __FLT_MAX_10_EXP__=38
  define: __FLT_MAX_EXP__=128
  define: __FLT_MAX__=3.40282347e+38F
  define: __FLT_MIN_10_EXP__=(-37)
  define: __FLT_MIN_EXP__=(-125)
  define: __FLT_MIN__=1.17549435e-38F
  define: __DBL_DENORM_MIN__=4.9406564584124654e-324
  define: __DBL_HAS_DENORM__=1
  define: __DBL_DIG__=15
  define: __DBL_DECIMAL_DIG__=17
  define: __DBL_EPSILON__=2.2204460492503131e-16
  define: __DBL_HAS_INFINITY__=1
  define: __DBL_HAS_QUIET_NAN__=1
  define: __DBL_MANT_DIG__=53
  define: __DBL_MAX_10_EXP__=308
  define: __DBL_MAX_EXP__=1024
  define: __DBL_MAX__=1.7976931348623157e+308
  define: __DBL_MIN_10_EXP__=(-307)
  define: __DBL_MIN_EXP__=(-1021)
  define: __DBL_MIN__=2.2250738585072014e-308
  define: __LDBL_DENORM_MIN__=3.64519953188247460253e-4951L
  define: __LDBL_HAS_DENORM__=1
  define: __LDBL_DIG__=18
  define: __LDBL_DECIMAL_DIG__=21
  define: __LDBL_EPSILON__=1.08420217248550443401e-19L
  define: __LDBL_HAS_INFINITY__=1
  define: __LDBL_HAS_QUIET_NAN__=1
  define: __LDBL_MANT_DIG__=64
  define: __LDBL_MAX_10_EXP__=4932
  define: __LDBL_MAX_EXP__=16384
  define: __LDBL_MAX__=1.18973149535723176502e+4932L
  define: __LDBL_MIN_10_EXP__=(-4931)
  define: __LDBL_MIN_EXP__=(-16381)
  define: __LDBL_MIN__=3.36210314311209350626e-4932L
  define: __POINTER_WIDTH__=64
  define: __BIGGEST_ALIGNMENT__=16
  define: __WINT_UNSIGNED__=1
  define: __INT8_TYPE__=signed char
  define: __INT8_FMTd__="hhd"
  define: __INT8_FMTi__="hhi"
  define: __INT8_C_SUFFIX__=
  define: __INT16_TYPE__=short
  define: __INT16_FMTd__="hd"
  define: __INT16_FMTi__="hi"
  define: __INT16_C_SUFFIX__=
  define: __INT32_TYPE__=int
  define: __INT32_FMTd__="d"
  define: __INT32_FMTi__="i"
  define: __INT32_C_SUFFIX__=
  define: __INT64_TYPE__=long int
  define: __INT64_FMTd__="ld"
  define: __INT64_FMTi__="li"
  define: __INT64_C_SUFFIX__=L
  define: __UINT8_TYPE__=unsigned char
  define: __UINT8_FMTo__="hho"
  define: __UINT8_FMTu__="hhu"
  define: __UINT8_FMTx__="hhx"
  define: __UINT8_FMTX__="hhX"
  define: __UINT8_C_SUFFIX__=
  define: __UINT8_MAX__=255
  define: __INT8_MAX__=127
  define: __UINT16_TYPE__=unsigned short
  define: __UINT16_FMTo__="ho"
  define: __UINT16_FMTu__="hu"
  define: __UINT16_FMTx__="hx"
  define: __UINT16_FMTX__="hX"
  define: __UINT16_C_SUFFIX__=
  define: __UINT16_MAX__=65535
  define: __INT16_MAX__=32767
  define: __UINT32_TYPE__=unsigned int
  define: __UINT32_FMTo__="o"
  define: __UINT32_FMTu__="u"
  define: __UINT32_FMTx__="x"
  define: __UINT32_FMTX__="X"
  define: __UINT32_C_SUFFIX__=U
  define: __UINT32_MAX__=4294967295U
  define: __INT32_MAX__=2147483647
  define: __UINT64_TYPE__=long unsigned int
  define: __UINT64_FMTo__="lo"
  define: __UINT64_FMTu__="lu"
  define: __UINT64_FMTx__="lx"
  define: __UINT64_FMTX__="lX"
  define: __UINT64_C_SUFFIX__=UL
  define: __UINT64_MAX__=18446744073709551615UL
  define: __INT64_MAX__=9223372036854775807L
  define: __INT_LEAST8_TYPE__=signed char
  define: __INT_LEAST8_MAX__=127
  define: __INT_LEAST8_FMTd__="hhd"
  define: __INT_LEAST8_FMTi__="hhi"
  define: __UINT_LEAST8_TYPE__=unsigned char
  define: __UINT_LEAST8_MAX__=255
  define: __UINT_LEAST8_FMTo__="hho"
  define: __UINT_LEAST8_FMTu__="hhu"
  define: __UINT_LEAST8_FMTx__="hhx"
  define: __UINT_LEAST8_FMTX__="hhX"
  define: __INT_LEAST16_TYPE__=short
  define: __INT_LEAST16_MAX__=32767
  define: __INT_LEAST16_FMTd__="hd"
  define: __INT_LEAST16_FMTi__="hi"
  define: __UINT_LEAST16_TYPE__=unsigned short
  define: __UINT_LEAST16_MAX__=65535
  define: __UINT_LEAST16_FMTo__="ho"
  define: __UINT_LEAST16_FMTu__="hu"
  define: __UINT_LEAST16_FMTx__="hx"
  define: __UINT_LEAST16_FMTX__="hX"
  define: __INT_LEAST32_TYPE__=int
  define: __INT_LEAST32_MAX__=2147483647
  define: __INT_LEAST32_FMTd__="d"
  define: __INT_LEAST32_FMTi__="i"
  define: __UINT_LEAST32_TYPE__=unsigned int
  define: __UINT_LEAST32_MAX__=4294967295U
  define: __UINT_LEAST32_FMTo__="o"
  define: __UINT_LEAST32_FMTu__="u"
  define: __UINT_LEAST32_FMTx__="x"
  define: __UINT_LEAST32_FMTX__="X"
  define: __INT_LEAST64_TYPE__=long int
  define: __INT_LEAST64_MAX__=9223372036854775807L
  define: __INT_LEAST64_FMTd__="ld"
  define: __INT_LEAST64_FMTi__="li"
  define: __UINT_LEAST64_TYPE__=long unsigned int
  define: __UINT_LEAST64_MAX__=18446744073709551615UL
  define: __UINT_LEAST64_FMTo__="lo"
  define: __UINT_LEAST64_FMTu__="lu"
  define: __UINT_LEAST64_FMTx__="lx"
  define: __UINT_LEAST64_FMTX__="lX"
  define: __INT_FAST8_TYPE__=signed char
  define: __INT_FAST8_MAX__=127
  define: __INT_FAST8_FMTd__="hhd"
  define: __INT_FAST8_FMTi__="hhi"
  define: __UINT_FAST8_TYPE__=unsigned char
  define: __UINT_FAST8_MAX__=255
  define: __UINT_FAST8_FMTo__="hho"
  define: __UINT_FAST8_FMTu__="hhu"
  define: __UINT_FAST8_FMTx__="hhx"
  define: __UINT_FAST8_FMTX__="hhX"
  define: __INT_FAST16_TYPE__=short
  define: __INT_FAST16_MAX__=32767
  define: __INT_FAST16_FMTd__="hd"
  define: __INT_FAST16_FMTi__="hi"
  define: __UINT_FAST16_TYPE__=unsigned short
  define: __UINT_FAST16_MAX__=65535
  define: __UINT_FAST16_FMTo__="ho"
  define: __UINT_FAST16_FMTu__="hu"
  define: __UINT_FAST16_FMTx__="hx"
  define: __UINT_FAST16_FMTX__="hX"
  define: __INT_FAST32_TYPE__=int
  define: __INT_FAST32_MAX__=2147483647
  Processing folder (recursive): /usr/lib/llvm-4.0/lib/clang/4.0.1/include/
  define: __INT_FAST32_FMTd__="d"
  define: __INT_FAST32_FMTi__="i"
  define: __UINT_FAST32_TYPE__=unsigned int
  define: __UINT_FAST32_MAX__=4294967295U
  define: __UINT_FAST32_FMTo__="o"
  define: __UINT_FAST32_FMTu__="u"
  define: __UINT_FAST32_FMTx__="x"
  define: __UINT_FAST32_FMTX__="X"
  define: __INT_FAST64_TYPE__=long int
  define: __INT_FAST64_MAX__=9223372036854775807L
  define: __INT_FAST64_FMTd__="ld"
  define: __INT_FAST64_FMTi__="li"
  define: __UINT_FAST64_TYPE__=long unsigned int
  define: __UINT_FAST64_MAX__=18446744073709551615UL
  define: __UINT_FAST64_FMTo__="lo"
  define: __UINT_FAST64_FMTu__="lu"
  define: __UINT_FAST64_FMTx__="lx"
  define: __UINT_FAST64_FMTX__="lX"
  define: __USER_LABEL_PREFIX__=
  define: __FINITE_MATH_ONLY__=0
  define: __GNUC_GNU_INLINE__=1
  define: __GCC_ATOMIC_TEST_AND_SET_TRUEVAL=1
  define: __GCC_ATOMIC_BOOL_LOCK_FREE=2
  define: __GCC_ATOMIC_CHAR_LOCK_FREE=2
  define: __GCC_ATOMIC_CHAR16_T_LOCK_FREE=2
  define: __GCC_ATOMIC_CHAR32_T_LOCK_FREE=2
  define: __GCC_ATOMIC_WCHAR_T_LOCK_FREE=2
  define: __GCC_ATOMIC_SHORT_LOCK_FREE=2
  define: __GCC_ATOMIC_INT_LOCK_FREE=2
  define: __GCC_ATOMIC_LONG_LOCK_FREE=2
  define: __GCC_ATOMIC_LLONG_LOCK_FREE=2
  define: __GCC_ATOMIC_POINTER_LOCK_FREE=2
  define: __NO_INLINE__=1
  define: __FLT_EVAL_METHOD__=0
  define: __FLT_RADIX__=2
  define: __DECIMAL_DIG__=__LDBL_DECIMAL_DIG__
  define: __amd64__=1
  define: __amd64=1
  define: __x86_64=1
  define: __x86_64__=1
  define: __k8=1
  define: __k8__=1
  define: __tune_k8__=1
  define: __REGISTER_PREFIX__=
  define: __NO_MATH_INLINES=1
  define: __FXSR__=1
  define: __SSE2__=1
  define: __SSE2_MATH__=1
  define: __SSE__=1
  define: __SSE_MATH__=1
  define: __MMX__=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4=1
  define: __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8=1
  define: __unix=1
  define: __unix__=1
  define: __linux=1
  define: __linux__=1
  define: __gnu_linux__=1
  define: __ELF__=1
  define: _GNU_SOURCE=1
  define: __FLOAT128__=1
  define: __STDC__=1
  define: __STDC_HOSTED__=1
  define: __cplusplus=201406L
  define: __STDCPP_DEFAULT_NEW_ALIGNMENT__=16UL
  define: __STDC_UTF_16__=1
  define: __STDC_UTF_32__=1
  other: --clang
  other: --clang_version=40001
  stdver: --c++17
  intelliSenseMode: clang
  Processing folder (recursive): /usr/local/include/
  Processing folder (recursive): /usr/include/
Checking for syntax errors: file:///home/allight/misc/compile-commands-example/test.cc
queue_update_intellisense for files in tu of: /home/allight/misc/compile-commands-example/test.cc
  Processing folder (recursive): /home/allight/misc/compile-commands-example/
  Discovering files: 4477 file(s) processed
  0 file(s) removed from database
Done discovering files.
Parsing open files...
Parsing remaining files...
  Parsing: 0 files(s) processed
Done parsing remaining files.
Done parsing open files.
textDocument/hover
terminating child process: 43357
errorSquiggles count: 0
textDocument/hover
textDocument/hover
```

Extension version: 0.21.0
VS Code version: Code 1.30.2 (61122f88f0bf01e2ac16bdb9e1bc4571755f5bd8, 2019-01-07T22:49:48.319Z)
OS version: Linux x64 4.18.10-1rodete2-amd64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) Gold 6154 CPU @ 3.00GHz (72 x 1200)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|187.62GB (2.05GB free)|
|Process Argv|.|
|Screen Reader|no|
|VM|0%|

</details>

**Comment by sean-mcmanus** [comment]
I got a repro on Linux. It gets permanently stuck trying to read results back from the non-existent compiler.

**Comment by allight** [comment]
> I got a repro on Linux. It gets permanently stuck trying to read results back from the non-existent compiler.

The compiler totally exists. It is at a relative path from the command-entries 'directory' (assuming you have a clang installed). Everything about that command is supposed to be interpreted as though that directory is the $PWD.

This bug was found when using VSCode on the AOSP repository. For AOSP builds are handled by prebuilt compilers distributed with the code repository (hence they have paths relative to the root of the project).

**Comment by sean-mcmanus** [comment]
Yeah, I meant to say it doesn't exist because of the bug which doesn't append the directory so it attempts to run a non-existent process. We'll try to release an Insiders fix Thursday.

**Comment by sean-mcmanus** [comment]
Relative paths for compilers in compile_command.json should be fixed with 0.22.0-insiders. You can get it via setting the C_Cpp.updateChannel setting to "Insiders", or downloading it manually from https://github.com/Microsoft/vscode-cpptools/releases .
